### PR TITLE
Support JSONC

### PR DESCRIPTION
**Pull request by ralfhandl**
Is just a special case of JSON5

Fixes https://github.com/ajv-validator/ajv-cli/issues/103

### CODE REVIEW

**Comment by Jason3S** [comment]
@epoberezkin Is there a reason to not take this pull request?

File: README.md [hunk 1]
```diff
@@ -201,36 +201,36 @@ This command supports the same options and parameters as [validate](#validate-da
...
-| `--messages=false`           | do not include text messages in errors |
-| `--loop-required=`           | max size of `required` to compile to expression (rather than to loop) |
-| `--loop-enum=`               | max size of `enum` to compile to expression (rather than to loop) |
-| `--own-properties`           | only validate own properties (not relevant for JSON, but can have effect for JavaScript objects) |
-| Code generation |
-| `--code-es5`                 | generate ES5 code |
-| `--code-lines`               | generate multi-line code |
-| `--code-optimize=`           | disable optimization (`false`) or number of optimization passes (1 pass by default) |
-| `--code-formats=`            | code to require formats object (only needed if you generate standalone code and do not use [ajv-formats](https://github.com/ajv-validator/ajv-formats)) |
+| Option                        | Description                                                                                                                                                                                            |
```
Comment (Jason3S) [inline README.md:204]: I only see formatting changes to this file.

File: src/commands/ajv.ts [hunk 2]
```diff
@@ -22,7 +22,11 @@ export default function (argv: ParsedArgs): AjvCore {
   if (invalid) process.exit(1)
   return ajv
 
-  function addSchemas(args: string | string[] | undefined, method: AjvMethod, fileType: string): void {
+  function addSchemas(
```
Comment (Jason3S) [inline src/commands/ajv.ts:25]: Also just formatting changes.

File: src/commands/compile.ts [hunk 3]
```diff
@@ -50,7 +50,7 @@ function execute(argv: ParsedArgs): boolean {
     const sch = openFile(file, `schema ${file}`)
     try {
       const id = sch?.$id
-      ajv.addSchema(sch, id ? undefined : file )
+      ajv.addSchema(sch, id ? undefined : file)
```
Comment (Jason3S) [inline src/commands/compile.ts:53]: The same for this file, just formatting changes.

File: src/commands/test.ts [hunk 4]
```diff
@@ -33,7 +33,9 @@ function execute(argv: ParsedArgs): boolean {
   const ajv = getAjv(argv)
   const validate = compile(ajv, argv.s)
   const shouldBeValid = !!argv.valid && argv.valid !== "false"
-  return getFiles(argv.d).map(testDataFile).every((x) => x)
+  return getFiles(argv.d)
```
Comment (Jason3S) [inline src/commands/test.ts:36]: This also just looks like formatting changes.

File: src/commands/util.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import path = require("path")
 import fs = require("fs")
 import yaml = require("js-yaml")
 import JSON5 = require("json5")
-import { AnyValidateFunction } from "ajv/dist/core"
+import {AnyValidateFunction} from "ajv/dist/core"
```
Comment (Jason3S) [inline src/commands/util.ts:7]: It is better to avoid formatting changes.

File: src/commands/validate.ts [hunk 6]
```diff
@@ -31,7 +31,9 @@ export default cmd
 function execute(argv: ParsedArgs): boolean {
   const ajv = getAjv(argv)
   const validate = compile(ajv, argv.s)
-  return getFiles(argv.d).map(validateDataFile).every((x) => x)
+  return getFiles(argv.d)
```
Comment (Jason3S) [inline src/commands/validate.ts:34]: Again formatting changes.

**Review by Jason3S — COMMENTED** [review]
Thank you for making the pull-request. I was thinking about doing the same thing.

I put a comment in each of the files that seem to only have formatting changes. Having a consistent format across all the files is nice to have, but they are unrelated to adding `jsonc` support. 

To make it easier for the maintainers of the repo to accept your pull request, I suggest limiting the changes to just the 3 files needed for the change. 

* `util.ts`
* `valid_data.jsonc`
* `validate.spec.ts`

Again thank your for make the effort to add `jsonc` support.

**Comment by ralfhandl** [comment]
@Jason3S Thanks! Actually the reformatting was done by this pre-commit hook, so I assume @epoberezkin wants them to be re-formatted 😄 

https://github.com/ajv-validator/ajv-cli/blob/d53d73ec1a393a11eaebcdeadd93fd393072bf4f/package.json#L72

**Comment by ralfhandl** [comment]
Fixed platform-dependency of `build` step

**Comment by ralfhandl** [comment]
@epoberezkin could you please consider merging this PR. Changes are now back to minimal.

**Comment by epoberezkin** [comment]
sorry for the delay, focus was clearly elsewhere...

**Comment by epoberezkin** [comment]
@ralfhandl what was the motivation for `.jsonc` to mean JSON5?

I should have checked, I don't see it anywhere in [JSON5 docs](https://json5.org) that it's a common extension for it...

Potentially, it should be a different option entirely to allow non-standard extensions to be interpreted as a specific format.

**Comment by ralfhandl** [comment]
@epoberezkin JSONC is [JSON with Comments](https://code.visualstudio.com/docs/languages/json#_json-with-comments) as for example used in VS Code configuration files.

Syntactically it is a subset of JSON5: just allowing `//` and `/* ... */` comments in addition to regular JSON.

Which is why using the JSON5 parser to read JSONC files works.

**Comment by epoberezkin** [comment]
Thank you!

**Pull request merged**